### DISCUSSION TITLE
Add search suggestions to OpenSearch description

### DISCUSSION THREAD

**Discussion by aminomancer** [npm/feedback#799] [Website]
npmjs.com has an [OpenSearch description](https://static.npmjs.com/osd.xml), which browsers can use to add a search engine for querying the registry. But it doesn't provide a search suggestion URL, which limits its utility. It wouldn't be very difficult to provide search suggestions, because the site already supports search suggestions (`https://www.npmjs.com/search/suggestions?q={searchString}`). Ordinarily, this could be added to a search engine manually using the WebExtensions [chrome_settings_overrides](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/chrome_settings_overrides) key, but it requires a different format for search suggestions. So here is my proposal:

Support a request like this: `https://www.npmjs.com/search/suggestions?q=test&format=opensearch`
Respond like this:
```jsonc
[
  "test",
  [
    "test",
    "test-exclude",
    "test-console",
    "@testing-library/react-hooks",
    "test-listen",
    "test-lwc-19",
    "@testing-library/user-event",
    "test-value",
    "testcafe",
    "@testlwc/test-lwc-19"
  ],
  ["", "", "", "", "", "", "", "", "", ""],
  [
    "https://www.npmjs.com/package/test",
    "https://www.npmjs.com/package/test-exclude",
    "https://www.npmjs.com/package/test-console",
    "https://www.npmjs.com/package/%40testing-library%2Freact-hooks",
    "https://www.npmjs.com/package/test-listen",
    "https://www.npmjs.com/package/test-lwc-19",
    "https://www.npmjs.com/package/%40testing-library%2Fuser-event",
    "https://www.npmjs.com/package/test-value",
    "https://www.npmjs.com/package/testcafe",
    "https://www.npmjs.com/package/%40testlwc%2Ftest-lwc-19"
  ]
]
```
Add this to the OpenSearch description on npmjs.com:
```udiff
--- a/osd.xml
+++ b/osd.xml
@@ -1,7 +1,8 @@
 <OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
   <ShortName>npm</ShortName>
   <Description>Search for packages on npm</Description>
   <InputEncoding>UTF-8</InputEncoding>
   <Image width="16" height="16" type="image/x-icon">https://static.npmjs.com/favicon-16x16.png</Image>
   <Url type="text/html" method="get"  template="https://www.npmjs.com/search?q={searchTerms}"/>
+  <Url type="application/x-suggestions+json" method="get" template="https://www.npmjs.com/search/suggestions?q={searchTerms}&format=opensearch" />
 </OpenSearchDescription>
```

I could file a PR if someone points me to the site repo. I couldn't find it so assumed it must not be public.

Thanks!

**Comment by YJL-NLP** [comment]
你好，我已经收到邮件，我会尽快处理。
